### ISSUE TITLE
Empty Interpreter Input/Output Buffer Pointers

### ISSUE THREAD

**Issue by embeddetech** [tensorflow/tensorflow#47452]
### 1. System information

- OS Platform and Distribution: Windows 10, building in Visual Studio
- TensorFlow installation: Model trained from PIP package (Tensorflow 2.4.1). TFLite interpreter building 2.0.0 (haven't been able to build TFLite 2.4.1 in Visual Studio yet)
- TensorFlow library: 2.4.1 (Train, conversion) and 2.0.0 (inference)

### 2. Code

My model uses the following layers: 
First stage, the stage I'm having problems:
TimeDistributed(Conv2D), TimeDistributed(MaxPooling2D), TimeDistributed(BatchNormalization), TimeDistributed(Flatten), TimeDistributed(Dense), 

Second stage (evaluated independently, haven't gotten to it yet):
Bidirectional(GRU), Reshape, Dense, Dropout.

I am quantizing this model. The inference code is shown below.

`	FS_FILE* pModelFile;
	char ModelStage1Filename[] = "\\Models\\cls_stg1.tflite";
	char ModelStage2Filename[] = "\\Models\\cls_stg2.tflite";
	int BytesRead;

	pModelFile = fs_fopen(ModelStage1Filename, "r");
	if (pModelFile != 0)
	{
		Stage1ModelSize = fs_fread(ModelStage1ModelBinary, 1, sizeof(ModelStage1ModelBinary), pModelFile);
		fs_fclose(pModelFile);
	}
	pModelFile = fs_fopen(ModelStage2Filename, "r");
	if (pModelFile != 0)
	{
		Stage1ModelSize = fs_fread(ModelStage2ModelBinary, 1, sizeof(ModelStage2ModelBinary), pModelFile);
		fs_fclose(pModelFile);
	}

	pMicroErrorReporter = new tflite::MicroErrorReporter();
	pErrorReporter = pMicroErrorReporter;

	if (Stage1ModelSize > 0)
	{
		Stage1Model = tflite::GetModel(ModelStage1ModelBinary);
		if (Stage1Model->version() != TFLITE_SCHEMA_VERSION)
		{
			/*TF_LITE_REPORT_ERROR(error_reporter,
				"Model provided is schema version %d not equal "
				"to supported version %d.\n",
				model->version(), TFLITE_SCHEMA_VERSION);*/
		}
		else
		{
			pStage1Interpreter = new tflite::MicroInterpreter(Stage1Model, resolver, stage_1_tensor_arena, stage_1_tensor_arena_size, pErrorReporter);
			if (pStage1Interpreter != 0)
			{
				pStage1Interpreter->AllocateTensors();
				// Obtain a pointer to the model's input tensor
				TfLiteTensor* input = pStage1Interpreter->input(0);
				if (input != 0)
				{
					if (input->dims->size == 4 &&
						input->type == kTfLiteUInt8 &&
						input->dims->data[0] == 1 &&
						input->dims->data[1] == 37 &&
						input->dims->data[2] == 256 &&
						input->dims->data[3] == 1)
					{
						TfLiteTensor* output = pStage1Interpreter->output(0);
						if (output != 0)
						{
							if (output->dims->size == 2 &&
								output->type == kTfLiteUInt8 &&
								output->dims->data[0] == 1 &&
								output->dims->data[1] == 64)
							{
								// All Good!
								Stage1Good = 1;
								pStage1Input = pStage1Interpreter->typed_input_tensor<unsigned char>(0);
								pStage1Output = pStage1Interpreter->typed_output_tensor<unsigned char>(0);
							}
						}
					}
				}
			}
		}
	}
	else
	{
	}`

### 3. Failure after conversion

My fully trained model is split into two parts, a non-RNN first stage and an RNN second stage. When I run the inference code above to prepare to do inference on the first stage, everything looks good, the input and output tensor shapes and sizes are correct. But executing the lines below, I get null pointers, so I am unable to perform inference.

pStage1Input = pStage1Interpreter->typed_input_tensor<unsigned char>(0);
pStage1Output = pStage1Interpreter->typed_output_tensor<unsigned char>(0);

When I use this code and setup with a simple model with a single dense layer, I get valid pointers. 

### 4. (optional) RNN conversion support
This first stage does not include the RNN (the Bidirectional(GRU) ) parts of the full model.

### 5. (optional) Any other info / logs
I have tried to pull in Tensorflow Lite 2.4.1 into my Visual Studio project, however I am currently unable to resolve all of the dependencies, as the Windows make process seemed to work before in version 2.0.0 but not now in 2.4.1. There is a failure getting flatbuffers. So I don't know for sure if there is an issue with the difference in training/converting in 2.4.1 and performing inference using 2.0.0.

**Comment by abattery** [comment]
@embeddetech The above code looks correct. Generally, there are two TF versions, one for training and conversion and one for inference. Could you make sure that the TF version used for executions should have the same TF version with the one for conversion or higher? We do not guarantee the compatibility that the TF for inference has a lower version than the TF for training and conversion.

**Comment by abattery** [comment]
Please take a look at https://www.tensorflow.org/lite/guide/build_cmake to build the recent TensorFlow Lite version under Windows 10.

**Comment by embeddetech** [comment]
Thank you @abattery, I wasn't having any luck with bazel or make, I will try this cmake method. I just need to be able to see it build in Windows 10 to make sure I have the correct dependencies, then I can pull it into Visual Studio.

Would it be expected to have those input and output tensor pointers be null if there were version compatibility issues? Should be no issues with the layers I mentioned above executing in TensorFlow Lite, right?

Thank you!

**Comment by abattery** [comment]
For bazel, you can build a .dll with ```bazel build -c opt //tensorflow/lite/c:tensorflowlite_c```.

**Comment by abattery** [comment]
TFLite 2.0.0 version is quite an old version, which was released more than one year ago to be supported. It would be better to see this problem is happening at the recent TF version.

**Comment by embeddetech** [comment]
OK, I'll try to get it upgraded. The first issue is just getting it to build with all source code properly in place. The second issue is getting it to build in Visual Studio with MSVC compiler. I believe I've come across two compiler incompatibilities, which I was able to resolve, things like MSVC not being happy about empty array initializers. I might do a pull request on those after confirming.

That bazel option, is that tested on Windows? I was trying a different target on Windows and it was choking on a flatbuffers package checksum error I believe.

**Comment by abattery** [comment]
If the above flatbuffer package checksum issue is still happening at the HEAD, please report it to us with the full log messages. I think it might be considered as a separate issue.

**Comment by embeddetech** [comment]
The CMake build _seemed_ to work, though I do get an error below.

`  quantize.cc
  random_standard_normal.cc
  random_uniform.cc
**C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.28.29333\include\random(1876,5): error C2338: invalid template argu
ment for uniform_int_distribution: N4659 29.6.1.1 [rand.req.genl]/1e requires one of short, int, long, long long, unsigned short, unsigned int, unsig
ned long, or unsigned long long [C:\Users\jtork\CMake\minimal_build\tensorflow-lite\tensorflow-lite.vcxproj]**
C:\Users\jtork\CMake\tensorflow_src\tensorflow\lite\kernels\random_uniform.cc(43): message : see reference to class template instantiation 'std::unif
orm_int_distribution<int8_t>' being compiled [C:\Users\jtork\CMake\minimal_build\tensorflow-lite\tensorflow-lite.vcxproj]
C:\Users\jtork\CMake\tensorflow_src\tensorflow\lite\kernels\random_uniform.cc(145): message : see reference to function template instantiation 'void
tflite::ops::custom::random_uniform::`anonymous-namespace'::RandomUniformSample<int8_t,std::uniform_int_distribution<int8_t>>(std::default_random_eng
ine &,T *,size_t,T,T)' being compiled [C:\Users\jtork\CMake\minimal_build\tensorflow-lite\tensorflow-lite.vcxproj]
          with
          [
              T=int8_t
          ]
**C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.28.29333\include\random(1876,1): error C2338: note: char, signed ch
ar, unsigned char, char8_t, int8_t, and uint8_t are not allowed [C:\Users\jtork\CMake\minimal_build\tensorflow-lite\tensorflow-lite.vcxproj]**
  range.cc
  rank.cc
  read_variable.cc
  Generating Code...
  Compiling...
  reduce.cc

`

After those errors, it seems to keep going. It finishes printing out all of the files and then stops, I don't get a clear sense of whether linking was happening or just compiling. How do I confirm that it built properly?

**Comment by abattery** [comment]
@embeddetech The above random uniform op is recently added and the error can be fixed by a simple fix. Will provide a fix with few days.

**Comment by embeddetech** [comment]
Thank you for your help, @abattery!

**Comment by embeddetech** [comment]
@abattery Do you know if CMake only works with Visual Studio Professional, not Community? I used Professional in the results above, however actually opening Visual Studio Pro and using it for a C# application, I was unable to get a valid license for some reason. So I uninstalled Professional and installed Community and started getting this error when trying to build:

```
C:\Users\jtork\CMake\tflite_build>cmake --build . -j
CMake Error:
  Generator

    Visual Studio 16 2019

  could not find specified instance of Visual Studio:

    C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional
```

I since uninstalled and reinstalled CMake and it still produces the same result....

**Comment by abattery** [comment]
@embeddetech Sorry. I do not know the details of the differences of the Visual Studio editions.

**Comment by embeddetech** [comment]
@abattery I found that when performing step 4 of the CMake build link you provided above, it checks for an installer and uses that from then on. After doing that step is when I uninstalled Pro and installed Community. After blowing everything away and starting over, I can see in that step 4 where it finds Community instead of Professional.

```
C:\Users\jtork\CMake\tflite_build>cmake ../tensorflow_src/tensorflow/lite
-- Building for: Visual Studio 16 2019
. . . 
**-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/cl.exe - skipped**
```

**Comment by embeddetech** [comment]
Just FYI, there are a few other suspicious strings emitted that may or may not be real errors during the execution of "CMake .../tensorflow_src/tensorflow/lite":

```
-- Looking for pthread.h
-- Looking for pthread.h - not found
```


```
-- Performing Test EIGEN_COMPILER_SUPPORT_CPP11
-- Performing Test EIGEN_COMPILER_SUPPORT_CPP11 - Failed
-- Performing Test COMPILER_SUPPORT_std=cpp03
-- Performing Test COMPILER_SUPPORT_std=cpp03 - Failed
```

```
-- Performing Test FARMHASH_HAS_BUILTIN_EXPECT
-- Performing Test FARMHASH_HAS_BUILTIN_EXPECT - Failed
```

**Comment by embeddetech** [comment]
@abattery I just attempted to run the make option for building a tensorflow lite for microcontrollers "Hello World" example on this fresh git pull. This resulted in the flatbuffers checksum error I mentioned before. This is when I try to run the following:

`make -f tensorflow/lite/micro/tools/make/Makefile TARGET=arc_emsdp OPTIMIZED_KERNEL_DIR=arc_mli ARC_TAGS=no_arc_mli generate_hello_world_make_project`

The results are below. Did you say I needed to post this as an issue in the main head?


```
FIND: Parameter format not correct
FIND: Parameter format not correct
--2021-02-28 17:27:56--  http://mirror.tensorflow.org/github.com/google/flatbuffers/archive/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip
Resolving mirror.tensorflow.org (mirror.tensorflow.org)... 142.250.113.128, 2607:f8b0:4023:1006::80
Connecting to mirror.tensorflow.org (mirror.tensorflow.org)|142.250.113.128|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1760478 (1.7M) [application/zip]
Saving to: '/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip'

/tmp/dca12522a9f9e37f126ab925 100%[=================================================>]   1.68M  9.18MB/s    in 0.2s

2021-02-28 17:27:57 (9.18 MB/s) - '/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip' saved [1760478/1760478]

/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 1: $'PK\003\004': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 2: nL8Q5: command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 3: nL8Q?: command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 4: $'\bnL8Qw\030B\005\352\002\304\005V': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 4: $'\367\257XY': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 5: $'\002T': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 5: $'v\355\236\215\265M4\036w]\027\262\036p\250t5\256\367\245f': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 5: $'\225L\347\313l\036\020\350\341\320\265\254\3218\002~\264B\323\302\371\016XC\2408\313': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 5: $'@\244\315\217\v\2105': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 5: $'\301\2173H2\037.\342,\311N\223\233d}\231^\257\341': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 5: $'^\255\342\345:\231g\220\256': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 5: $'m\237\220Q7\037\324\242\304R\234R\372T\235\317k\347T\347\316\334\235\353F\352\217g\317q': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 5: 6□m@0□cs/d□□ □j☼□4□E)8□&□□□z@-i#hPo□q□→☻X□6□□
□l□□▼{□□A♂y+□□□→□□□Ff□‼%□□3⌂2□□܇□5□s□□D%→k<□Ȃ§J□□□↔W♣□9□►□0□ub♦OO□□□_□ɕ□@□□Y→□0□!□↓□m□□b□Ѫh□R□(K□H□2@L□90j□□63J□^%,□G□□j-\□□↓♦☺liW□□□A♀□27/□1□□□∟U□e☺'□j♀Yk§□u9□□□#□3□(lඤ□d7□□□Z4K▬□Ez□□y}□]O□□,□m⌂'>!□(□}□□□□►□W□t§□□⌂|^□□□8□□Zu□↕□∟v{♫□♠□Jt+{ߐ%^□□k<~.□ـ□Z46♦□□v9□□   ↕□□v□□Z□Ԋ□ݏܲ{7□oȽ♦□⌂)♀‼H\□/     □e▼□C□w□□@□□□□□☼□□B2□#-{♥□□0□□□□□l>9�□t)□□s↓↑_□□P2□     PK♥♦
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 11: $'OO\337\367\312\262\204\0365\3341P]\034\327\025ZL4\237t\364]r\363\024c\020\202\216\214?h\210\254N\326\b\a\275\221z]P\022\301c\177\346\2608\274\350\234\3230tl4y\006\244\272h.y\371\214i\201,': command not found
nL8Q>   flatbuffers-dca12522a9f9e37f126ab925fd385c807ab4f84e/.bazelci/UT♣☺□□l_PK♥♦
nL8Q=□!X□K      flatbuffers-dca12522a9f9e37f126ab925fd385c807ab4f84e/.bazelci/presubmit.ymlUT♣☺□□l_□□□□J*□□I□L□L-□R□I,I-.□*□i□E□□V
□I□y%□□f♠& □□☻X}|IbQzjI1DHWAIOOO        □♠↓□K▬j□♣§□□ML□/□□¶PK♥♦
nL8Q□4%_□0☺B    flatbuffers-dca12522a9f9e37f126ab925fd385c807ab4f84e/.clang-formatUT♣☺□□l_m□Mn□@♀□□>♣↨)[v►□□*RP□♣\p♠+□y□n□Q⌂□E□: No such file or directory
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 11: $'\034Z\303\211\346\236\223\377\b\277\211\215l\270\247\326\b\257ot[\005\235:\232\004\235\243\246\277\231\304m\242\3711\277\356\361\235$\235\264\031Y\363\351\2365\363\334\356\004\343\232-_\346_\332+\355[\264\371\021\324J\234\256\377B\252\252\002\370\004PK\003\004': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 12: $'\bnL8Q\33088\322\236\337B': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 13: $'\302@\020D\373': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 16: $'\021\301\336\316\322?\020': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 16: r□□□□▲T↑□bLl□XN□□□QtK*◄4: No such file or directory
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 16: $'\202\v%Q\256\253\333\341^\211ulh': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 17: $'f\345eey\206\310?S\336\253wZ\270pF\223\304\2026\306\3236\375{\351\003PK\003\004': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 18: $'\bnL8Q\324\240P3\211\375A': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: command substitution: line 19: syntax error near unexpected token `)'/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: command substitution: line 19: `☻1♀D□□□□▬▬□□□□⌂□▲d□mJ□□.□□[□^ĝ3o↕=☻&♦9bxB □0j□<□□□□k$□♦↨x)(b"1 □□P□x□□l'
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 16: $'\305\202\205\270\312\0261\315\274\304RT\373t\363\206\036t\371\261\204d\340V\327\257v\230\036\236\332\300U0\330_\246\345\f\343L\336\027\027mg\252\333\256\230\235g\376\201I\275\317\037PK\003\004': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 17: $'\bnL8Q\026\345\365\275OUC': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 18: $'\200': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 18: $'\365\370\243\371PK\003\004': command not found
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 19: flatbuffers-dca12522a9f9e37f126ab925fd385c807ab4f84e/.github/UT♣☺□□l_PK♥♦: No such file or directory
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 20: syntax error near unexpected token `('
/tmp/dca12522a9f9e37f126ab925fd385c807ab4f84e.zip: line 20: nL8Q}□□□B☻N flatbuffers-dca12522a9f9e37f126ab925fd385c807ab4f84e/.github/ISSUE_TEMPLATE.mdUT♣☺□□l_U□AO□0♀□□□§o□□(□□v@∟ABp☻  ♫H□♫^□□□Ҥ□]`□▲□C□I□!□□□=□□□□□!□hr□□□>□□Ԃ↕□□□♂□^#□0zڳU¶□□C□q□↓□1↕□,□□|□□a□\#RjGj□B.+□□☼!r□'↨     9͏y□B□□∟D□?KR□□↑I□GkX□$g□‼B9☻B□F^9□□♣A□q►□□□u□□□¶□1□□□bj□↓1□=HO□□□□□□,□□□!▲-□9□G□s□→☼□□□W□      □=□L□□□□<\]Uh□□zu[□□▼□□□□□□□S□7□dvT@↑JּeY□      ∟□      q)Z□□r□□□♦□;□□򑱡 ↑□♠,□٠nV□□sϿ□□6□□□□{4=;□I♫%□▲□f□□□□☺PK♥♦'
tensorflow/lite/micro/tools/make/Makefile:525: *** Something went wrong with the flatbuffers download: Bad checksum. Expected: aa9adc93eb9b33fa1a2a90969e48baee, Got: .  Stop.
```

**Comment by abattery** [comment]
I managed to resolve the random uniform related error in https://github.com/tensorflow/tensorflow/commit/358948ec8da338b694416b4cd83d5ed3648933b0. Could you try again? For the flatbuffer checksum issue, can you create a separate issue in the github to track?

**Comment by embeddetech** [comment]
Will do, will notify you shortly to confirm the fix. Added Issue [47467](https://github.com/tensorflow/tensorflow/issues/47467) for the flatbuffers issue.

**Comment by embeddetech** [comment]
Steps:

- Open command prompt
- Navigate to CMake installation directory
- Run git clone https://github.com/tensorflow/tensorflow.git tensorflow_src
- Run "mkdir tflite_build" to create tflite_build folder
- Run "cd tflite_build" to go into that folder
- Run "cmake ../tensorflow_src/tensorflow/lite". Appears successful, however potential issues listed under "Build Config Results" below.
- Run "cmake --build . -j" to build Tensorflow Lite. This appears to build correctly, I see tensorflow-lite.lib/pdb under the tflite_build/Debug folder. Lots of warnings produced, listed under "Library Build Warnings" below.

If the issues/warnings below are not a concern, I'll be working to pull this TensorflowLite source code into Visual Studio and attempt to perform inference on my model and see if I get valid pointers from the inference object.


**Build Config Results:**
```
-- Performing Test EIGEN_COMPILER_SUPPORT_CPP11
-- Performing Test EIGEN_COMPILER_SUPPORT_CPP11 - Failed
-- Found Threads: TRUE
-- Performing Test EIGEN_COMPILER_SUPPORT_CPP11
-- Performing Test EIGEN_COMPILER_SUPPORT_CPP11 - Failed
-- Performing Test COMPILER_SUPPORT_std=cpp03
-- Performing Test COMPILER_SUPPORT_std=cpp03 - Failed
```

```
-- Performing Test FARMHASH_HAS_BUILTIN_EXPECT
-- Performing Test FARMHASH_HAS_BUILTIN_EXPECT - Failed
```


**Library Build Warnings:**

```

  c_api.cc
C:\Users\jtork\CMake\tensorflow_src\tensorflow\lite\c\c_api.cc(88,29): warning C4273: 'TfLiteVersion': inconsistent dll linkage [C:\Users\jtork\CMake\tflite_build\ten
sorflow-lite.vcxproj]
C:\Users\jtork\CMake\tensorflow_src\tensorflow/lite/c/c_api.h(96,36): message : see previous definition of 'TfLiteVersion' [C:\Users\jtork\CMake\tflite_build\tensorfl
ow-lite.vcxproj]
C:\Users\jtork\CMake\tensorflow_src\tensorflow\lite\c\c_api.cc(90,75): warning C4273: 'TfLiteModelCreate': inconsistent dll linkage [C:\Users\jtork\CMake\tflite_build
\tensorflow-lite.vcxproj]
C:\Users\jtork\CMake\tensorflow_src\tensorflow/lite/c/c_api.h(99,37): message : see previous definition of 'TfLiteModelCreate' [C:\Users\jtork\CMake\tflite_build\tens
orflow-lite.vcxproj]
C:\Users\jtork\CMake\tensorflow_src\tensorflow\lite\c\c_api.cc(97,64): warning C4273: 'TfLiteModelCreateFromFile': inconsistent dll linkage [C:\Users\jtork\CMake\tfli
te_build\tensorflow-lite.vcxproj]
C:\Users\jtork\CMake\tensorflow_src\tensorflow/lite/c/c_api.h(103,37): message : see previous definition of 'TfLiteModelCreateFromFile' [C:\Users\jtork\CMake\tflite_b
uild\tensorflow-lite.vcxproj]

[Cutting out a lot of these messages]

C:\Users\jtork\CMake\tflite_build\eigen\unsupported\Eigen\CXX11\src/ThreadPool/ThreadLocal.h(277): message : see reference to class template instantiation 'EigenForTF
Lite::TensorEvaluator<ArgType,Device>::EvalParallelContext<DoneCallback,false,false,false,0>::ThreadLocalBlocksInitialize<float *,true>' being compiled [C:\Users\jtor
k\CMake\tflite_build\tensorflow-lite.vcxproj]
          with
          [
              ArgType=const EigenForTFLite::TensorContractionOp<const std::array<EigenForTFLite::IndexPair<EigenForTFLite::DenseIndex>,1>,const EigenForTFLite::Tensor
  ReshapingOp<const EigenForTFLite::DSizes<__int64,2>,const EigenForTFLite::TensorImagePatchOp<-1,-1,const EigenForTFLite::TensorMap<EigenForTFLite::Tensor<const floa
  t,4,1,EigenForTFLite::DenseIndex>,16,EigenForTFLite::MakePointer>>>,const EigenForTFLite::TensorReshapingOp<const EigenForTFLite::DSizes<__int64,2>,const tflite::mu
  ltithreaded_ops::ConstEigenTensor>,const EigenForTFLite::NoOpOutputKernel>,
              Device=EigenForTFLite::ThreadPoolDevice,
              DoneCallback=EigenForTFLite::TensorEvaluator<const EigenForTFLite::TensorContractionOp<const std::array<EigenForTFLite::IndexPair<EigenForTFLite::DenseI
  ndex>,1>,const EigenForTFLite::TensorReshapingOp<const EigenForTFLite::DSizes<__int64,2>,const EigenForTFLite::TensorImagePatchOp<-1,-1,const EigenForTFLite::Tensor
  Map<EigenForTFLite::Tensor<const float,4,1,EigenForTFLite::DenseIndex>,16,EigenForTFLite::MakePointer>>>,const EigenForTFLite::TensorReshapingOp<const EigenForTFLit
  e::DSizes<__int64,2>,const tflite::multithreaded_ops::ConstEigenTensor>,const EigenForTFLite::NoOpOutputKernel>,EigenForTFLite::ThreadPoolDevice>::NoCallback
          ]
C:\Users\jtork\CMake\tflite_build\eigen\unsupported/Eigen/CXX11/src/Tensor/TensorContractionThreadPool.h(803): message : see reference to class template instantiation
 'EigenForTFLite::ThreadLocal<EigenForTFLite::TensorEvaluator<ArgType,Device>::EvalParallelContext<DoneCallback,false,false,false,0>::ThreadLocalBlocks<float *>,Eigen
ForTFLite::TensorEvaluator<ArgType,Device>::EvalParallelContext<DoneCallback,false,false,false,0>::ThreadLocalBlocksInitialize<float *,true>,EigenForTFLite::TensorEva
luator<ArgType,Device>::EvalParallelContext<DoneCallback,false,false,false,0>::ThreadLocalBlocksRelease<float *>>' being compiled [C:\Users\jtork\CMake\tflite_build\t
ensorflow-lite.vcxproj]
          with
          [
              ArgType=const EigenForTFLite::TensorContractionOp<const std::array<EigenForTFLite::IndexPair<EigenForTFLite::DenseIndex>,1>,const EigenForTFLite::Tensor
  ReshapingOp<const EigenForTFLite::DSizes<__int64,2>,const EigenForTFLite::TensorImagePatchOp<-1,-1,const EigenForTFLite::TensorMap<EigenForTFLite::Tensor<const floa
  t,4,1,EigenForTFLite::DenseIndex>,16,EigenForTFLite::MakePointer>>>,const EigenForTFLite::TensorReshapingOp<const EigenForTFLite::DSizes<__int64,2>,const tflite::mu
  ltithreaded_ops::ConstEigenTensor>,const EigenForTFLite::NoOpOutputKernel>,
              Device=EigenForTFLite::ThreadPoolDevice,
              DoneCallback=EigenForTFLite::TensorEvaluator<const EigenForTFLite::TensorContractionOp<const std::array<EigenForTFLite::IndexPair<EigenForTFLite::DenseI
  ndex>,1>,const EigenForTFLite::TensorReshapingOp<const EigenForTFLite::DSizes<__int64,2>,const EigenForTFLite::TensorImagePatchOp<-1,-1,const EigenForTFLite::Tensor
  Map<EigenForTFLite::Tensor<const float,4,1,EigenForTFLite::DenseIndex>,16,EigenForTFLite::MakePointer>>>,const EigenForTFLite::TensorReshapingOp<const EigenForTFLit
  e::DSizes<__int64,2>,const tflite::multithreaded_ops::ConstEigenTensor>,const EigenForTFLite::NoOpOutputKernel>,EigenForTFLite::ThreadPoolDevice>::NoCallback
          ]
		  
		  
  minimal_logging_default.cc
C:\Users\jtork\CMake\tensorflow_src\tensorflow\lite\minimal_logging_default.cc(28,9): warning C4068: unknown pragma 'clang' [C:\Users\jtork\CMake\tflite_build\tensorf
low-lite.vcxproj]
C:\Users\jtork\CMake\tensorflow_src\tensorflow\lite\minimal_logging_default.cc(29,9): warning C4068: unknown pragma 'clang' [C:\Users\jtork\CMake\tflite_build\tensorf
low-lite.vcxproj]
C:\Users\jtork\CMake\tensorflow_src\tensorflow\lite\minimal_logging_default.cc(31,9): warning C4068: unknown pragma 'clang' [C:\Users\jtork\CMake\tflite_build\tensorf
low-lite.vcxproj]


```

**Comment by abattery** [comment]
The build config test is used for finding out the correct set of the options for the given compilers. They are just indicators of the MSVC's characteristics.

For the compiler warnings, TFLite has a number of other libraries including eigen and so on. It is common to see warnings during compliations.

**Comment by embeddetech** [comment]
I am working on trying to manually pull in the appropriate Tensorflow Lite code into a Visual C project from the CMake build as a reference. I am floundering a bit, as there is just so much code. A working make build of one of the Tensorflow Lite for Microcontrollers examples would really help clear the clutter (that's what I used in the past). I'm running Windows, so I'd have to get up to speed on Docker, I believe, to see a make build of one of the Tensorflow Lite for Microcontrollers examples build. Unless the make issues for Windows were resolved.

My options appear to be:

1. Go through the tedious work of trying to manually pull the source code from the CMake build of Tensorflow Lite into Visual Studio and work through issues. No idea how long this would take.
2. Do a make build of a Tensorflow Lite for Microcontrollers example in Docker, and use that as a reference for pulling in minimal source code into Visual Studio.
3. Pray that the Windows make build for the Tensorflow Lite for Microcontrollers gets fixed, then use that as a reference for pulling in minimal source code into Visual Studio.

Once I get the latest and greatest pulled into VS, I can see if my null pointer issues are resolved and I can perform inference with my model.
